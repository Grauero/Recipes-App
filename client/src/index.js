import React, { Fragment, lazy, Suspense } from 'react';
import PropTypes from 'prop-types';
import ReactDOM from 'react-dom';
import {
  BrowserRouter as Router, Route, Switch, Redirect
} from 'react-router-dom';
import ApolloClient from 'apollo-boost';
import { ApolloProvider } from 'react-apollo';
import * as serviceWorker from './serviceWorker';

import App from './components/App';
import Navbar from './components/layouts/Navbar';
import Spinner from './components/common/Spinner';
import withSession from './components/utils/withSession';
import './index.css';

const Search = lazy(() => import('./components/recipe/Search'));
const Signin = lazy(() => import('./components/auth/Signin'));
const Signup = lazy(() => import('./components/auth/Signup'));
const AddRecipe = lazy(() => import('./components/recipe/AddRecipe'));
const RecipePage = lazy(() => import('./components/recipe/RecipePage'));
const Profile = lazy(() => import('./components/profile/Profile'));

const client = new ApolloClient({
  uri: process.env.REACT_APP_GRAPHQL_SERVER,
  fetchOptions: {
    credentials: true
  },
  request: (operation) => {
    const token = localStorage.getItem('token');

    operation.setContext({
      headers: {
        authorization: token
      }
    });
  },
  onError: ({ networkError }) => {
    if (networkError) {
      console.log('Network Error', networkError);
    }
  }
});

const Root = ({ refetch, session }) => (
  <Router>
    <Fragment>
      <Navbar session={session} />
      <Switch>
        <Route exact path="/" component={App} />
        <Route path="/search" component={Search} />
        <Route path="/signin" render={() => <Signin refetch={refetch} />} />
        <Route path="/signup" render={() => <Signup refetch={refetch} />} />
        <Route path="/recipe/add" render={() => <AddRecipe session={session} />} />
        <Route path="/recipes/:_id" component={RecipePage} />
        <Route path="/profile" render={() => <Profile session={session} />} />
        <Redirect to="/" />
      </Switch>
    </Fragment>
  </Router>
);

Root.propTypes = {
  refetch: PropTypes.func.isRequired,
  session: PropTypes.instanceOf(Object).isRequired
};

const RootWithSession = withSession(Root);

ReactDOM.render(
  <ApolloProvider client={client}>
    <Suspense fallback={<div className="fallback"><Spinner /></div>}>
      <RootWithSession />
    </Suspense>
  </ApolloProvider>,
  document.getElementById('root')
);

serviceWorker.unregister();
