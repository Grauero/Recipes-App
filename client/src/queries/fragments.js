import { gql } from 'apollo-boost';

const recipeFragments = {
  recipe: gql`
    fragment CompleteRecipe on Recipe {
      _id
      name
      imageUrl
      category
      description
      instructions
      createdDate
      likes
      username
    }
  `,
  like: gql`
    fragment LikeRecipe on Recipe {
      _id
      likes
    }
  `
};

export default recipeFragments;
