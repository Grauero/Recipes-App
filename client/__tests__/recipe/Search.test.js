import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import toJSON from 'enzyme-to-json';
import { MockedProvider } from 'react-apollo/test-utils';

import Search from '../../src/components/recipe/Search';
import { SEARCH_RECIPES } from '../../src/queries';

const initialState = { searchResults: [] };

const mocks = [
  {
    request: {
      query: SEARCH_RECIPES,
      variables: { searchTerm: 'searchTerm' }
    },
    result: {
      data: {
        searchRecipes: { result: 'result' }
      }
    }
  }
];
let component;
let wrapper;

beforeEach(() => {
  wrapper = mount(
    <MockedProvider mocks={mocks} addTypename={false}>
      <BrowserRouter>
        <Search />
      </BrowserRouter>
    </MockedProvider>
  );

  component = wrapper.find(Search);
  component.setState(initialState);
});

afterEach(() => {
  wrapper.unmount();
});

it('matches snapshot', () => {
  expect(toJSON(component)).toMatchSnapshot();
});
