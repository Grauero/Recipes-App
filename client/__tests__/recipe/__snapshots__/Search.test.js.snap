// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Search>
  <ApolloConsumer>
    <div
      className="App"
    >
      <input
        className="search"
        onChange={[Function]}
        placeholder="Search for Recipes"
        type="search"
      />
      <ul />
    </div>
  </ApolloConsumer>
</Search>
`;
